### PR TITLE
Code review config center

### PR DESCRIPTION
**Pull request by beiwei30**
## What is the purpose of the change

XXXXX

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/config/AbstractPrefixConfiguration.java [hunk 1]
```diff
@@ -45,7 +45,10 @@ public Object getProperty(String key, Object defaultValue) {
             value = getInternalProperty(prefix + key);
         }
         if (value == null) {
```
Comment (khanimteyaz) [inline dubbo-common/src/main/java/org/apache/dubbo/common/config/AbstractPrefixConfiguration.java]: Can be simplified with 
```
if (value == null) {
     value = getInternalProperty(key);
 }
 return value!=null? value:defaultValue;
```

File: dubbo-common/src/main/java/org/apache/dubbo/common/config/Configuration.java [hunk 2]
```diff
@@ -67,7 +73,15 @@
      * @return the value to which this configuration maps the specified key, or default value if the configuration
      * contains no mapping for this key.
      */
-    Object getProperty(String key, Object defaultValue);
+    default Object getProperty(String key, Object defaultValue) {
+        Object value = getInternalProperty(key);
+        if (value == null) {
+            value = defaultValue;
+        }
+        return value;
```
Comment (khanimteyaz) [inline dubbo-common/src/main/java/org/apache/dubbo/common/config/Configuration.java]: can be simplified with 
`return value!=null? value : defaultValue?`

File: dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfiguration.java [hunk 3]
```diff
@@ -98,57 +94,47 @@ protected void initWith(URL url) {
...
-            key = group + "/" + key;
-        }
-        // when group is null, we are fetching governance rules.
-        // for example, key=org.apache.dubbo.DemoService.configurators
-        else {
-            int i = key.lastIndexOf(".");
-            key = key.substring(0, i) + "/" + key.substring(i + 1);
+    public Object getInternalProperty(String key) {
+        ChildData childData = treeCache.getCurrentData(key);
+        if (childData != null) {
```
Comment (khanimteyaz) [inline dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfiguration.java]: can be simplified with 
`return childData!=null ?return new String(childData.getData(), StandardCharsets.UTF_8) :null;`

File: dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfiguration.java [hunk 4]
```diff
@@ -98,57 +94,47 @@ protected void initWith(URL url) {
...
+    public String getConfig(String key, String group, long timeout) throws IllegalStateException {
+        // when group is not null, we are getting startup configs from Config Center
+        // for example, group=dubbo, key=dubbo.properties
+        if (StringUtils.isNotEmpty(group)) {
+            key = group + "/" + key;
         }
-        return null;
+        // when group is null, we are fetching governance rules.
+        // for example, key=org.apache.dubbo.DemoService.configurators
+        else {
```
Comment (khanimteyaz) [inline dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfiguration.java]: is there any possibility of key having null?

File: dubbo-common/src/main/java/org/apache/dubbo/common/config/AbstractPrefixConfiguration.java (hunk 1 above)
Comment (chickenlj) [reply dubbo-common/src/main/java/org/apache/dubbo/common/config/AbstractPrefixConfiguration.java]: done

File: dubbo-common/src/main/java/org/apache/dubbo/common/config/Configuration.java (hunk 2 above)
Comment (chickenlj) [reply dubbo-common/src/main/java/org/apache/dubbo/common/config/Configuration.java]: done

File: dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfiguration.java (hunk 4 above)
Comment (chickenlj) [reply dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfiguration.java]: In reality, key passed to here will never be null.

File: dubbo-common/src/main/java/org/apache/dubbo/common/config/CompositeConfiguration.java [hunk 5]
```diff
@@ -56,7 +56,7 @@ public void addConfiguration(int pos, Configuration configuration) {
     }
 
     @Override
-    protected Object getInternalProperty(String key) {
+    public Object getInternalProperty(String key) {
         Configuration firstMatchingConfiguration = null;
         for (Configuration config : configList) {
             try {
```
[the remark this answers is not in the document]
Comment (chickenlj) [reply dubbo-common/src/main/java/org/apache/dubbo/common/config/CompositeConfiguration.java]: Changed the behaviour to print log.

**Pull request merged**